src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ResourceBundle\Controller;
+
+use Hateoas\Configuration\Route;
+use Hateoas\Representation\Factory\PagerfantaFactory;
+use Pagerfanta\Pagerfanta;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * @author Paweł Jędrzejewski <pawel@sylius.org>
+ */
+class ResourcesCollectionProvider implements ResourcesCollectionProviderInterface
+{
+    /**
+     * @var PagerfantaFactory
+     */
+    private $pagerfantaRepresentationFactory;
+
+    /**
+     * @param PagerfantaFactory $pagerfantaRepresentationFactory
+     */
+    public function __construct(PagerfantaFactory $pagerfantaRepresentationFactory)
+    {
+        $this->pagerfantaRepresentationFactory = $pagerfantaRepresentationFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get(RequestConfiguration $requestConfiguration, RepositoryInterface $repository)
+    {
+        if (null !== $factoryMethod = $requestConfiguration->getRepositoryMethod(null)) {
```

Comment:
I think this `$requestConfiguration->getRepositoryMethod(null)` is same as `$requestConfiguration->getRepositoryMethod()`